### PR TITLE
Collect inf about replicas & group from all nodes

### PR DESCRIPTION
**Pull request by otrach**
Fixes ZPS-7465.

* Collect info about replicas and group from all Windows nodes on which Availability Group is placed.
* Remove modeling of some Always On properties which then obtain during monitoring.
* Some small fixes and refactoring.

### CODE REVIEW

File: ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py [hunk 1]
```diff
@@ -74,7 +74,7 @@
     'powershell MSSQL Job',
     'powershell MSSQL AO AG',
     'powershell MSSQL AO AR',
-    'powershell MSSQL AO AL',
+    'powershell AO AL',
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py:77]: As [here](https://github.com/zenoss/ZenPacks.zenoss.Microsoft.Windows/blob/cff575638d3d6b0f65d059304768f12a92753007/ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py#L1544) all strategies which starts with 'powershell MSSQL' use SQL Connection, and Availability Listener strategy doesn't require SQL Connection, so  I renamed it accordingly.

File: ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py [hunk 2]
```diff
@@ -1028,7 +1025,7 @@ def parse_result(config, result):
 class PowershellMSSQLAlwaysOnALStrategy(object):
     implements(IStrategy)
 
-    key = 'MSSQLAlwaysOnAL'
+    key = 'AlwaysOnAL'
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py:1028]: Case described [here](https://github.com/zenoss/ZenPacks.zenoss.Microsoft.Windows/pull/1150/files#r575250280)

File: ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py [hunk 3]
```diff
@@ -1141,7 +1138,7 @@ def parse_result(config, result):
         return parsed_results
 
 
-gsm.registerUtility(PowershellMSSQLAlwaysOnALStrategy(), IStrategy, 'powershell MSSQL AO AL')
+gsm.registerUtility(PowershellMSSQLAlwaysOnALStrategy(), IStrategy, 'powershell AO AL')
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py:1141]: Case described [here](https://github.com/zenoss/ZenPacks.zenoss.Microsoft.Windows/pull/1150/files#r575250280)

File: ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py [hunk 4]
```diff
@@ -1398,7 +1395,7 @@ def params(cls, datasource, context):
                                         'powershell MSSQL Job,'
                                         'powershell MSSQL AO AG',
                                         'powershell MSSQL AO AR',
-                                        'powershell MSSQL AO AL',
+                                        'powershell AO AL',
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py:1398]: Case described [here](https://github.com/zenoss/ZenPacks.zenoss.Microsoft.Windows/pull/1150/files#r575250280)

File: ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py [hunk 5]
```diff
@@ -1577,6 +1572,9 @@ def collect(self, config):
                 command_line, script = strategy.build_command_line(cmd_line_input, db_indices, counters)
             else:
                 command_line, script = strategy.build_command_line(cmd_line_input)
+        elif dsconf0.params['strategy'] == 'powershell AO AL':
+            cmd_line_input = dsconf0.params.get('instanceid', '')  # Take listener ID as input parameter.
+            command_line, script = strategy.build_command_line(cmd_line_input)
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py:1577]: Case described [here](https://github.com/zenoss/ZenPacks.zenoss.Microsoft.Windows/pull/1150/files#r575250280)

File: ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py [hunk 6]
```diff
@@ -814,39 +808,43 @@ def parse_result(config, result):
...
+            sql_instance_data = {
+                'quorum_state': lookup_ag_quorum_state(
+                    ag_results.get('cl_quorum_state'))
+            }
+            ag_om = fill_ag_om(ag_om, ag_info_maps, prepId, sql_instance_data)
+            parsed_results['maps'].append(ag_om)
+
             # Events:
             # 1. check whether Primary replica SQL Instance has changed. If yes - create an event.
-            if ag_om:
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py:834]: No need this check as we check [this](https://github.com/zenoss/ZenPacks.zenoss.Microsoft.Windows/pull/1150/files#diff-e0d2db946d6c9d028116c3f0691aa319a3046026b95b3f3abd89da1ad7a21d05R834)

File: ZenPacks/zenoss/Microsoft/Windows/modeler/plugins/zenoss/winrm/WinMSSQL.py [hunk 7]
```diff
@@ -293,88 +294,56 @@ def __init__(self, conn_info, log):
         "  $ag_info['id'] = $ag_uid;"
         "  $ag_info['ag_res_id'] = $ag_res_uid;"
         "  $ag_info['primary_replica_server_name'] = $pr_rep_servr_name;"
-        # Put databases on top beacause we get them regardless of Replica role (either on Primary or Secondary Replicas)
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/modeler/plugins/zenoss/winrm/WinMSSQL.py:296]: Database related script block I moved on the bottom of the script as now we collect info from all Windows nodes on which Availability Group is placed.

File: ZenPacks/zenoss/Microsoft/Windows/modeler/plugins/zenoss/winrm/WinMSSQL.py [hunk 8]
```diff
@@ -293,88 +294,56 @@ def __init__(self, conn_info, log):
...
-        "   $ags.Add($ag_info) > $null;"
-        "   continue;"
-        "  }"
         "  $ag_info['is_distributed'] = $ag.IsDistributedAvailabilityGroup;"
         "  $ag_info['health_check_timeout'] = $ag.HealthCheckTimeout;"
         "  $ag_info['automated_backup_preference'] = $ag.AutomatedBackupPreference;"
         "  $ag_info['failure_condition_level'] = $ag.FailureConditionLevel;"
         "  $ag_info['cluster_type'] = $ag.ClusterTypeWithDefault;"
         "  $ag_info['db_level_health_detection'] = $ag.DatabaseHealthTrigger;"
-        "  $ag_uid = $ag.UniqueId;"
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/modeler/plugins/zenoss/winrm/WinMSSQL.py:329]: Remove modeling of some Always On properties which then obtain during monitoring.

File: ZenPacks/zenoss/Microsoft/Windows/modeler/plugins/zenoss/winrm/WinMSSQL.py [hunk 9]
```diff
@@ -293,88 +294,56 @@ def __init__(self, conn_info, log):
...
-
+        # Availability Replicas
         "  foreach ($ar in $ag.AvailabilityReplicas) {"
         "   $ar_inf = New-Object 'System.Collections.Generic.Dictionary[string, object]';"
         "   $ar_uid = $ar.UniqueId;"
         "   $ar_inf['ag_id'] = $ag_uid;"
         "   $ar_inf['ag_res_id'] = $ag_res_uid;"
         "   $ar_inf['id'] = $ar_uid;"
         "   $ar_inf['name'] = $ar.Name;"
-        "   $ar_inf['state'] = $ar.MemberState;"
```
Comment (otrach) [inline ZenPacks/zenoss/Microsoft/Windows/modeler/plugins/zenoss/winrm/WinMSSQL.py:352]: Remove modeling of some Always On properties which then obtain during monitoring.

**Review by idubei — APPROVED** [review]

**Pull request merged**